plugin/evm/block_builder.go

```diff
@@ -104,10 +107,13 @@ func (b *blockBuilder) awaitSubmittedTxs() {
 		extraChan = b.extraMempool.SubscribePendingTxs()
 	}
 
+	events := make(chan core.NewTxPoolReorgEvent)
+	sub := b.txPool.SubscribeNewReorgEvent(events)
```

Comment:
Should we be concerned around any races on the initialization of the two block hashes?

We call `awaitSubmittedTxs` when we are being marked as out of bootstrapping. So isn't it possible that we missed the reorg event but we are going to have set preference called afterwards?

If this happens wouldn't we never get a reorg event and deadlock?

obv this isn't a concern on mainnet (as someone else will build a block and then we'll make progress)... But it still seems potentially off.